src/games/supported/DemonAttack.cpp

```diff
@@ -121,3 +129,41 @@ void DemonAttackSettings::loadState(Deserializer & ser) {
   m_lives = ser.getInt();
 }
 
+// returns a list of mode that the game can be played in
+ModeVect DemonAttackSettings::getAvailableModes(){
+    ModeVect modes;
+    modes.push_back(1);
+    modes.push_back(3);
+    modes.push_back(5);
+    modes.push_back(7);
+
+    return modes;
+}
+
+// set the mode of the game
+// the given mode must be one returned by the previous function
+void DemonAttackSettings::setMode(game_mode_t m, System &system, StellaEnvironment& environment){
+    if(m==1 || m==3 || m==5 || m==7){
```

Comment:
All right, I think we should move to C++11. It's overdue now and there's a number of features here that will be much nicer in C++11. @mhauskn , @mcmachado , can you think of a reason not to move to C++11?

It probably makes sense to have a separate PR first moving ALE 0.5.1 to C++11 (in the dev branch), then continue the work here. WDYT?